packages/next/src/server/lib/utils.ts

```diff
@@ -16,45 +16,112 @@ const parseNodeArgs = (args: string[]) => {
 
   // For the `NODE_OPTIONS`, we support arguments with values without the `=`
   // sign. We need to parse them manually.
-  let found = null
+  let orphan = null
   for (let i = 0; i < tokens.length; i++) {
     const token = tokens[i]
 
     if (token.kind === 'option-terminator') {
       break
     }
 
-    // If we haven't found a possibly orphaned option, we need to look for one.
-    if (!found) {
-      if (token.kind === 'option' && typeof token.value === 'undefined') {
-        found = token
-      }
+    // When we encounter an option, if it's value is undefined, we should check
+    // to see if the following tokens are positional parameters. If they are,
+    // then the option is orphaned, and we can assign it.
+    if (token.kind === 'option') {
+      orphan = typeof token.value === 'undefined' ? token : null
+      continue
+    }
 
+    // If the token isn't a positional one, then we can't assign it to the found
+    // orphaned option.
+    if (token.kind !== 'positional') {
+      orphan = null
       continue
     }
 
-    // If the next token isn't a positional value, then it's truly orphaned.
-    if (token.kind !== 'positional' || !token.value) {
-      found = null
+    // If we don't have an orphan, then we can skip this token.
+    if (!orphan) {
       continue
     }
 
-    // We found an orphaned option. Let's add it to the values.
-    values[found.name] = token.value
-    found = null
+    // If the token is a positional one, and it has a value, so add it to the
+    // values object. If it already exists, append it with a space.
+    if (orphan.name in values && typeof values[orphan.name] === 'string') {
+      values[orphan.name] += ` ${token.value}`
+    } else {
+      values[orphan.name] = token.value
+    }
   }
 
   return values
 }
 
+/**
+ * Tokenizes the arguments string into an array of strings, supporting quoted
+ * values and escaped characters.
```

Comment:
```suggestion
 * values and escaped characters.
 * Converted from: https://github.com/nodejs/node/blob/c29d53c5cfc63c5a876084e788d70c9e87bed880/src/node_options.cc#L1401
```

Is it worth including where this code comes from? Do we need to include a license reference to Node.js?